test/easyconfigs/easyconfigs.py

```diff
@@ -1507,6 +1508,12 @@ def template_easyconfig_test(self, spec):
                     failing_checks.append("OpenSSL should not be listed as OS dependency")
 
     src_cnt = len(ec['sources'])
+
+    # if easyblock derives from Cargo easyblock we need to take into account list of crates,
+    # which get added to list of sources, to correctly determine number of checksums for sources (incl. crates)
+    if isinstance(app, Cargo):
+        src_cnt += len(ec.get('crates', []))
```

Comment:
Can't we use the actual list of sources as will be done by the checksums step?
I.e. `app.cfg['sources']` as discussed in the other PR/issue

Reply:
Not since the change in https://github.com/easybuilders/easybuild-easyblocks/pull/3448, because `ec['sources']` no longer contains also the crates since that change, that's exactly why this fix is required

Reply:
Since that change `app.cfg['sources`]` should be populated with the crates while `ec['sources']` should not. Or did I miss anything? See the [comment](https://github.com/easybuilders/easybuild-easyblocks/pull/3448#issuecomment-2351656529)
> `x = app_class(EasyConfig(file_content))`
> `x.cfg <-- always do this`

Reply:
I created an alternative PR: https://github.com/easybuilders/easybuild-easyconfigs/pull/21419

Taking the sources, patches and checksums from `app.cfg` there and in a single place.